### PR TITLE
spbase: Changes and comments from review meeting

### PR DESCRIPTION
**Pull request by DLWoodruff**
This has the spbase that resulted from the meeting.

### CODE REVIEW

File: mpisppy/spbase.py [hunk 1]
```diff
@@ -239,12 +248,11 @@ def create_scenarios(self, cb_data):
...
         """
         if self.scenarios_constructed:
-            if self.rank == self.rank0:
-                print("Warning: scenarios already constructed")
-            return
+            raise RuntimeError("Scenarios already constructed.")
 
         for sname in self.local_scenario_names:
             instance_creation_start_time = time.time()
+            ### s = self.scenario_creator(sname, **scenario_creator_kwargs)
```
Comment (bknueven) [inline mpisppy/spbase.py:255]: I thought we had made this change -- I'll open an issue.

File: mpisppy/cylinders/spcommunicator.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ def __init__(self, spbase_object, fullcomm, intercomm, intracomm, options=None):
         self.fullcomm = fullcomm
         self.intercomm = intercomm
         self.intracomm = intracomm
-        self.rank_global = fullcomm.Get_rank()
+        self.global_rank = fullcomm.Get_rank()
         self.rank_inter = intercomm.Get_rank()
```
Comment (dtmildebrath) [inline mpisppy/cylinders/spcommunicator.py:34]: Maybe we need to change `rank_intra` and `rank_inter` to `cylinder_rank` and `strata_rank`?

File: mpisppy/utils/sputils.py [hunk 3]
```diff
@@ -145,9 +145,9 @@ def make_comms(n_spokes, fullcomm=None):
 
     # Create the intercomm and intracomm
     # Cryptic comment: intra is vertical, inter is around the hub
-    rank_global = fullcomm.Get_rank()
-    intercomm = fullcomm.Split(key=rank_global, color=rank_global // nsp1)
-    intracomm = fullcomm.Split(key=rank_global, color=rank_global % nsp1)
+    global_rank = fullcomm.Get_rank()
+    intercomm = fullcomm.Split(key=global_rank, color=global_rank // nsp1)
+    intracomm = fullcomm.Split(key=global_rank, color=global_rank % nsp1)
     return intercomm, intracomm
```
Comment (dtmildebrath) [inline mpisppy/utils/sputils.py:151]: Similarly here, maybe `strata_comm` and `cylinder_comm`?

File: mpisppy/spbase.py [hunk 4]
```diff
@@ -97,28 +92,27 @@ def __init__(
...
-        self.look_before_leap_all()
-        self.compute_unconditional_node_probabilities()
-        self.attach_nlens()
-        self.attach_nonant_indexes()
-        self.attach_varid_to_nonant_index()
-        self.create_communicators()
-        self.set_sense()
-        self.set_multistage()
-        self.use_variable_probability_setter()
+        self._create_scenarios(cb_data)
```
Comment (dtmildebrath) [inline mpisppy/spbase.py:101]: Where did we land on removing `cb_data`?

Comment (bknueven) [reply mpisppy/spbase.py:101]: Yes, I opened an issue to track that (#66). I think we should do it as part of a separate PR.

File: mpisppy/cylinders/spcommunicator.py (hunk 2 above)
Comment (bknueven) [reply mpisppy/cylinders/spcommunicator.py]: Fixed

**Pull request merged**